src/app/static/src/js/components/runWorkflow/workflowSummary/runWorkflowSummaryHeader.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+<section>
+    <h2 id="summary-header">Summary</h2>
+    <div id="summary-warning" class="row mt-3" v-if="hasMissingDependencies">
+        <div class="col-auto">
+            <alert-triangle-icon size="2x" stroke="red" class="custom-class"/>
+        </div>
+        <div class="col-auto">
+            <span class="d-inline-block pb-2">Some reports depend on the latest version of other reports that are not included in your workflow:</span>
+            <div class="missingDependency" v-for="(missing_dependencies, report) in workflowSummary.missing_dependencies" :key="report">
+                <span v-if="missing_dependencies.length" class="font-weight-bold"> {{ report }}</span>
+                <ul v-for="missing_dependency in missing_dependencies" class="styled"  :key="missing_dependency">
+                    <li>{{ missing_dependency }}</li>
+                </ul>
+            </div>
+        </div>
+    </div>
+</section>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import {WorkflowSummaryResponse} from "../../../utils/types";
+import {AlertTriangleIcon} from "vue-feather-icons"
+
+interface Props {
+    workflowSummary: WorkflowSummaryResponse | null
+}
+
+interface Computed {
+    hasMissingDependencies: boolean;
+}
+export default Vue.extend<unknown, Computed, unknown, Props>({
+    name: "runWorkflowSummaryHeader",
+        components: {
+            AlertTriangleIcon
+        },
+    props: {
+        workflowSummary: {
+            type: Object || null,
+            default: null
+        }
+    },
+    computed: {
+        hasMissingDependencies() {
+            return this.workflowSummary && !!Object.keys(this.workflowSummary.missing_dependencies)
```

Comment:
Why the double negation? Also I think it would be nicer to avoid `Object.keys` here - could just pass the report list as a prop (the `reports` property of the `WorkflowSummary` object in the parent component) so that this becomes 

```
this.reports.some(report => this.workflowSummary.missing_dependencies[report.name].length > 0)
```

Reply:
Since `reports` is already available as a prop inside `workflowSummary`, I changed to the below. The double negation converts the possible null value to false.
`!!this.workflowSummary?.reports.some(report => this.workflowSummary.missing_dependencies[report.name].length > 0)`

Reply:
Even better. Double negation makes sense now I see the `?.`